src/transformStateWithClones.js

```diff
@@ -7,7 +7,41 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
-}
+  const arrayOfStates = [];
+  const stateCopy = { ...state };
+
+  for (const obj of actions) {
+    if (obj.type === 'clear') {
+      for (const stateKey in stateCopy) {
+        if (stateKey) {
+          delete stateCopy[stateKey];
+        }
+      }
+
+      const clearedCopy = { ...stateCopy };
+
+      arrayOfStates.push(clearedCopy);
```

Comment:
this is repeated so can be moved outside the `switch`

Reply:
made switch block instead of 'if'